src/controllers/dynakube/activegate/internal/tenantinfo/reconciler_test.go

```
@@ -0,0 +1,97 @@
+package tenantinfo
+
+import (
+	"context"
+	"testing"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/dtclient"
+	"github.com/Dynatrace/dynatrace-operator/src/scheme"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	testDynakubeName = "test-dynakube"
+	testNamespace    = "test-namespace"
+)
+
+var (
+	tenantInfoResponse = &dtclient.ActiveGateTenantInfo{
+		TenantInfo: dtclient.TenantInfo{
+			UUID:  "testUUID",
+			Token: "dt.some.valuegoeshere",
+		},
+		Endpoints: "someEndpoints",
+	}
+)
+
+func newTestReconciler(client client.Client) *Reconciler {
+	instance := &dynatracev1beta1.DynaKube{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: testNamespace,
+			Name:      testDynakubeName,
+		},
+		Spec: dynatracev1beta1.DynaKubeSpec{
+			APIURL: "https://testing.dev.dynatracelabs.com/api",
+		},
+	}
+	dtc := &dtclient.MockDynatraceClient{}
+	dtc.On("GetActiveGateTenantInfo", mock.Anything).Return(tenantInfoResponse, nil)
+
+	r := NewReconciler(client, client, scheme.Scheme, instance, dtc)
+	return r
+}
+
+func TestReconcile(t *testing.T) {
+	t.Run(`reconcile tenant info for first time`, func(t *testing.T) {
+		r := newTestReconciler(fake.NewClientBuilder().Build())
+		update, err := r.Reconcile()
+
+		var tenantInfoSecret corev1.Secret
+		r.Client.Get(context.TODO(), client.ObjectKey{Name: extendWithAGSecretSuffix(r.dynakube.Name), Namespace: testNamespace}, &tenantInfoSecret)
```

same as bevore, do not implicitly ignore an error